demos/gesture_recognition_demo/cpp_gapi/main.cpp

```diff
@@ -0,0 +1,236 @@
+
+bool ParseAndCheckCommandLine(int argc, char *argv[]) {
+    /** ---------- Parsing and validating input arguments ----------**/
+    gflags::ParseCommandLineNonHelpFlags(&argc, &argv, true);
+    if (FLAGS_h) {
+        showUsage();
+        showAvailableDevices();
+        return false;
+    }
+    slog::info << "Parsing input parameters" << slog::endl;
+    if (FLAGS_i.empty())
+        throw std::logic_error("Parameter -i is not set");
+    if (FLAGS_o.empty())
+        throw std::logic_error("Parameter -o is not set");
+    if (FLAGS_m_a.empty())
+        throw std::logic_error("Parameter -m_a is not set");
+    if (FLAGS_m_d.empty())
+        throw std::logic_error("Parameter -m_d is not set");
+    if (FLAGS_c.empty())
+        throw std::logic_error("Parameter -c is not set");
+
+    return true;
+}
+
+namespace nets {
+G_API_NET(PersoneDetection, <cv::GMat(cv::GMat)>, "perspne_detection");
+G_API_NET(ActionRecognition, <cv::GMat(cv::GMat)>, "action_recognition");
+}
+
+int main(int argc, char *argv[]) {
+    try {
+        if (!ParseAndCheckCommandLine(argc, argv)) {
+            return 0;
+        }
+
+        /** Get information about nets **/
+        const auto pd_net_shape = getNetShape(FLAGS_m_d);
+        const auto ar_net_shape = getNetShape(FLAGS_m_a);
+
+        /** Get information about frame from cv::VideoCapture **/
+        std::shared_ptr<ImagesCapture> cap = openImagesCapture(FLAGS_i, FLAGS_loop, 0,
+            std::numeric_limits<size_t>::max(), stringToSize(FLAGS_res));
+        const auto tmp = cap->read();
+        cap.reset();
+        if (!tmp.data) {
+            throw std::runtime_error("Couldn't grab first frame");
+        }
+        cv::Size frame_size = cv::Size{tmp.cols, tmp.rows};
+        cap = openImagesCapture(FLAGS_i, FLAGS_loop, 0,
+            std::numeric_limits<size_t>::max(), stringToSize(FLAGS_res));
+
+        /** Runtime key wrapper **/
+        std::shared_ptr<int> current_person_id = std::make_shared<int>(0);
+
+        /** ---------------- Main graph of demo ---------------- **/
+        /** Graph inputs **/
+        cv::GMat batch;
+        cv::GOpaque<std::shared_ptr<int>> selected_person_id;
+
+        cv::GMat fast_frame = custom::GetFastFrame::on(batch, frame_size);
+
+        /** Person detection **/
+        cv::GMat detections = cv::gapi::infer<nets::PersoneDetection>(fast_frame);
+
+        /** Get ROIs from detections **/
+        cv::GArray<TrackedObject> objects = custom::BoundingBoxExtract::on(detections, fast_frame, pd_net_shape);
+
+        /** Track detection **/
+        cv::GArray<TrackedObject> tracked = custom::PersonTrack::on(fast_frame, objects);
+
+        /** Create clip for AR net **/
+        cv::GArray<cv::GMat> clip = custom::ConstructClip::on(batch, tracked, selected_person_id, ar_net_shape, frame_size);
+
+        /** Action recognition **/
+        cv::GArray<cv::GMat> actions = cv::gapi::infer2<nets::ActionRecognition>(batch, clip);
```

Comment:
If `batch` is not used by this network (and it is not - since no ROIs passed), I'd pass `last_frame` instead

Reply:
however, maybe passing `batch` is ok too from the logical standpoint (as `clip` is created from the batch).

Reply:
Also, can `clip` be batch itself to run it via plain `infer` with no lists?